packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -120,42 +195,48 @@ def reduce_logging():
 
 @pytest.mark.skipif(not lb_enabled(), reason='Load Balancer disabled')
 def test_vip(dcos_api_session, reduce_logging):
-    """Test every permutation of VIP
-    """
+    '''Test every permutation of VIP
+    '''
     addrs = [['1.1.1.{}:{}', '1.1.1.{}:{}'],
              ['/namedvip{}:{}', 'namedvip{}.marathon.l4lb.thisdcos.directory:{}']]
-    samehosts = [True, False]
-    vipnets = ['USER', 'BRIDGE', 'HOST']
-    proxynets = ['USER', 'BRIDGE', 'HOST']
-    permutations = [[vi, va, sh, vn, pn]
+    # tests
+    # UCR doesn't support BRIDGE mode
+    permutations = [[c, vi, va, sh, vn, pn]
+                    for c in ['NONE', 'UCR', 'DOCKER']
                     for [vi, va] in addrs
-                    for sh in samehosts
-                    for vn in vipnets
-                    for pn in proxynets]
-    tests = [VipTest(i, vi, va, sh, vn, pn) for i, [vi, va, sh, vn, pn] in enumerate(permutations)]
-    threads = []
+                    for sh in [True, False]
+                    for vn in ['USER', 'BRIDGE', 'HOST']
+                    for pn in ['USER', 'BRIDGE', 'HOST']]
+    tests = [VipTest(i, c, vi, va, sh, vn, pn) for i, [c, vi, va, sh, vn, pn] in enumerate(permutations)]
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers=maxthreads)
     # deque is thread safe
     failed_tests = deque(tests)
     passed_tests = deque()
     skipped_tests = deque()
+    # skip certain tests
     for r in tests:
+        if r.container is 'UCR' or r.container is 'NONE':
+            if r.vipnet is 'BRIDGE' or r.proxynet is 'BRIDGE':
```

Comment:
Well, you have misused `is`. :D The operators are well-defined, and the purpose of `is` is limited to a _single_ thing.